// components/basketBar/index.js
import React, {Component} from "react";
import "./basket.css";
import {getTotalBasketCount, getTotalPrice} from "selectors";
import {connect} from "react-redux";
import {deleteItemOnBasket} from "actions";
import ReactModal from "react-modal";

class BasketBar extends Component {
    state = {
        showModal: false
    };

    onShowModal = () => {
        this.setState({
            showModal: !this.state.showModal
        })
    };

    render() {
        const {totalBasketCount, totalPrice, basket, deleteItemOnBasket} = this.props;
        return (

            <div className="cart btn-group dropdown">
                <button type="button" className="btn btn-default dropdown-toggle" data-toggle="dropdown"
                        aria-haspopup="true" aria-expanded="false">
                    <div className="containerForBasket">
                        <div>
                            <i className="fa fa-shopping-cart fa-3x"/>
                        </div>
                        <div className="containerForItems">
                            <span>Товаров: {totalBasketCount}</span>
                            <span>Сумма: {totalPrice}$</span>
                        </div>
                        <span className="caret"/>
                    </div>
                </button>
                <ul className="dropdown-menu dropdown-primary" aria-labelledby="dLabel">
                    {basket.length > 0 ? basket.map((item, id) => {
                            return (
                                <li key={id}>
                                    <span className="cardName">{item.cardName}   </span>
                                    <span>{item.quantity}шт.</span>
                                    <button className="btnDeleteItem" onClick={() => deleteItemOnBasket(item.cardId)}>
                                        <i className="fa fa-times" aria-hidden="true"/>
                                    </button>
                                </li>
                            )
                        })
                        : <li>Корзина пуста</li>}
                    <li role="separator" className="divider"/>
                    <li className="totalBasket">
                        <div>
                            <p>Общее количество: {totalBasketCount}</p>
                            <p>Общая сумма: {totalPrice}$</p>
                            <button type="button" className="btn btn-default pull-right btnPay"
                                    onClick={this.onShowModal}>Оплатить
                            </button>
                        </div>
                    </li>
                </ul>
                {this.state.showModal &&
                <ReactModal
                    isOpen={this.state.showModal}
                    contentLabel="Процесс оплаты"
                    style={{width: "400px"}}
                    ariaHideApp={false}>
                    <header>
                        <p>Процесс оплаты</p>
                        <button onClick={this.onShowModal}>
                            <i className="fa fa-times" aria-hidden="true"/>
                        </button>
                    </header>
                    <div>
                        {basket.length > 0 ? basket.map((item, id) => {
                            return (
                                <li className="listModal" key={id}>
                                    <div className="wrapModal">
                                        <div className="cardNameModal">{item.cardName}</div>
                                        <div>({item.quantity})</div>
                                        <div>{item.cardPrice * item.quantity}$</div>
                                    </div>
                                    <div className="pay">Олата прошла успешно!</div>
                                </li>
                            )
                        }) : <div className="pay">Выберите товары для оплаты</div>}
                    </div>
                </ReactModal>}
            </div>

        )
    }
}

{/*<ReactModal*/
}
{/*isOpen={this.state.showModal}*/
}
{/*contentLabel="Minimal Modal Example"*/
}
{/*>*/
}
{/*<button onClick={this.handleCloseModal}>Close Modal</button>*/
}
{/*</ReactModal>*/
}
const mapStateToProps = state => ({
    totalBasketCount: getTotalBasketCount(state),
    totalPrice: getTotalPrice(state),
    basket: state.basket
});

const mapDispatchToProps = {
    deleteItemOnBasket
};


export default connect(mapStateToProps, mapDispatchToProps)(BasketBar);
