// containers/items/index.js
import React, {Component} from "react";
import {connect} from "react-redux";
import {fetchItems} from "actions";
import {getItems} from "selectors";
import {Link} from "react-router";

class Items extends Component {

    componentDidMount() {
        this.props.fetchItems()
    }

    renderItem(item, index) {
        return (
            <div className="col-sm-4 col-lg-4 col-md-4 book-list" key={index}>
                <div className="caption">
                    <Link to={`/card/${item.id}`}>
                        <h4 className="pull-right">{item.title}</h4>
                    </Link>
                </div>
            </div>
        )
    }

    isEmpty(obj) {
        return Object.keys(obj).length === 0;
    }

    render() {
        const {items, params} = this.props;
        const loadItems = !this.isEmpty(items);
        const categoryId = params.id;
        return (
            <div>
                <div className="books row">
                    {loadItems && items[categoryId].map((item, index) => this.renderItem(item, index))}
                </div>
            </div>
        )
    }
}

const MapStateToProps = state => ({
    items: getItems(state)
});

const MapDispatchToProps = {
    fetchItems
};

export default connect(MapStateToProps, MapDispatchToProps)(Items);
