// containers/card/index.js
import React, {Component} from "react";
import {connect} from "react-redux";
import {fetchCard, addItemOnBasket} from "actions";
import Counter from "../../components/counter";
import "./card.css";


class Card extends Component {

    componentDidMount() {
        this.props.fetchCard(this.props.params.id);
    }

    newAddItem() {
        const {card, count} = this.props;
        let obj = {};
        obj.cardId = card.id;
        obj.quantity = count.value;
        obj.cardPrice = card.price;
        obj.cardName = card.title;
        return obj;
    }

    render() {
        const {card, addItemOnBasket} = this.props;
        const maxQuantity = card.quantity;
        const description = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam volutpat, nibh at luctus iaculis, elit diam sollicitudin felis, vulputate ultricies turpis urna nec turpis. Nam quis diam nec tellus ullamcorper efficitur sed at felis. Curabitur sit amet luctus tellus, sit amet gravida sem. Quisque eu justo ac risus scelerisque feugiat vel a tellus. Nulla ut venenatis dui, at dignissim dui. In sed rhoncus est. Pellentesque eget pellentesque est. Nunc rhoncus leo in justo pulvinar malesuada. Suspendisse fermentum luctus dignissim. Fusce porta facilisis nisl ac egestas. Sed vitae ante vel justo aliquet blandit. Morbi quam arcu, convallis id venenatis vel, ultrices non est. Praesent pretium hendrerit sodales.";
        return (
                <div className="container">
                    <div className="wrapper">
                        <div>
                            <img src="http://lorempixel.com/640/320/technics/"/>
                        </div>
                        <div className="buy-bar">
                            <Counter maxQuantity={maxQuantity}/>
                            <button className="btn" onClick={() => addItemOnBasket(this.newAddItem())}>Купить</button>
                        </div>

                    </div>
                    <div className="title">
                        <h3>{card.title}</h3>
                        <p className="lead">${card.price}</p>
                        {description}
                    </div>

                </div>
        )
    }
}

const MapStateToProps = state => ({
    card: state.card,
    count: state.count
});

const MapDispatchToProps = {
    fetchCard,
    addItemOnBasket
};

export default connect(MapStateToProps, MapDispatchToProps)(Card);
